File: src/main/java/xupt/se/ttms/idao/DAOFactory.java
package xupt.se.ttms.idao;

import xupt.se.ttms.dao.StudioDAO;

public class DAOFactory
{
    private static StudioDAO stuDao;

    public static synchronized StudioDAO creatStudioDAO()
    {
        if(null == stuDao)
            stuDao=new StudioDAO();
        return stuDao;
    }
}

File: src/main/java/xupt/se/ttms/idao/DAOFactory.kt
package xupt.se.ttms.idao;

import xupt.se.ttms.dao.EmployeeDAO
import xupt.se.ttms.dao.PlayDAO
import xupt.se.ttms.dao.StudioDAO

object DAOFactory {
    val stuDao: StudioDAO by lazy {
        StudioDAO()
    }
    val employeeDAO: EmployeeDAO by lazy {
        EmployeeDAO()
    }
    val playDAO: PlayDAO by lazy { PlayDAO() }
}